testsuite/tests/frame-pointers/fp_backtrace.c

```diff
@@ -108,10 +108,11 @@ static void print_symbol(const char* symbol, const regmatch_t* match)
 void fp_backtrace(value argv0)
 {
   const char* execname = String_val(argv0);
-  struct frame_info* next = NULL;
   const char* symbol = NULL;
 
-  for (struct frame_info* fi = __builtin_frame_address(0); fi; fi = next) {
+  for (struct frame_info* fi = __builtin_frame_address(0), *next = NULL;
```

Comment:
> I think you should either use `struct frame_info *fi = ..., *next = ...`

You mean in the scope above the loop, right?

We're doing clever things elsewhere… like traversing an array and counting the indices at the same time!

```c
value l;
intnat i;
/* ... */
for (i = 0, l = al; l != Val_emptylist; l = Field(l, 1), i++) {
  arrays[i] = Field(l, 0);
  offsets[i] = 0;
  lengths[i] = caml_array_length(Field(l, 0));
}
```

I'll try to harmonize walking over lists.

Reply:
Iterating on two things in parallel is okay, my comment was specifically about the use of `type* foo` for the first argument and `type *bar` for the second.

> I'll try to harmonize walking over lists.

Please don't mix too many changes in the current PR, every non-automatic change you add makes it noticeably harder to review. If it was just automatic changes it would be easier, manual changes need a different review approach and they don't mix well.